source/parser.hera

```diff
@@ -386,7 +386,15 @@ UnaryWithoutParenthesizedAssignmentBody
 
 UnaryPostfix
   QuestionMark
-  ( __ ( As / Satisfies ) Type )+ -> { ts: true, children: $0 }
+  TypePostfix+ -> { ts: true, children: $0 }
+
+TypePostfix
+  _:ws As:as ExclamationPoint?:ex Type:type ->
+    if (ex) {
+      return [" as unknown", ws, as, type]
```

Comment:
Not sure it's worth it, but `" as unknown"` could take `$loc` from the ExclamationPoint.